flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/optimize/RelNodeBlock.scala

```diff
@@ -352,7 +353,14 @@ class RelNodeBlockPlanBuilder private (tableConfig: ReadableConfig) {
 
 object RelNodeBlockPlanBuilder {
 
-  // It is a experimental config, will may be removed later.
+  /**
+   * Whether to treat union-all node as a breakpoint.
+   * @deprecated
+   *   This configuration has been deprecated as part of FLIP-457. Please use
+   *   [[org.apache.flink.table.api.config.OptimizerConfigOptions.TABLE_OPTIMIZER_UNIONALL_AS_BREAKPOINT_ENABLED]]
+   *   instead.
+   */
+  @Deprecated
```

Comment:
IIUC, the proposed change 'Move the configurations to ...' of the FLIP means that we can just make the move without having to keep the old positions because we haven't changed these configuration option names, so If there are no objections, you can remove the option from the original classes.

Reply:
> IIUC, the proposed change 'Move the configurations to ...' of the FLIP means that we can just make the move without having to keep the old positions because we haven't changed these configuration option names, so If there are no objections, you can remove the option from the original classes.

Yes, it's the original plan. But during the [discussion](https://lists.apache.org/thread/n52smchwgr6qfg3tfvmxpdshk8obgor2), Xuannan mentioned the API compatibility guarantees. The only concern is that although there's no constraint on the Experimental API, and in theory, users should not directly rely on table-planner in their projects, we're still determining whether users rely on these variables in their code.

Reply:
The cautious approach makes sense. In addition to marking it deprecated we'd better also add a note that it will be removed in version 2.0.